app/assets/javascripts/controllers/container_dashboard/util/dashboard-utils-factory.js

```diff
@@ -97,8 +97,28 @@ angular.module('miq.util').factory('dashboardUtilsFactory', function() {
   var parseDate = function(date) {
     myDate = Date.parse(date);
     return isNaN(myDate) ? date : myDate
-  }
+  };
+  var iconClassForProvider = function(provider) {
```

Comment:
> would it make sense to merge this logic with container topology getIcon? Seems like we have multiple mechanisms for "give me an icon for this thing" in containers JS, so I'm thinking they should be kept together. (We do have a few more on non-container ruby code, but still...) (But that one deals with unicode codes instead of class names so.. I don't know, just something to think about)

Uh, I think theyre different enough to keep apart but Im not keen on whats happening inside topology. What do you think @abonas ?